pkg/ast/spl/spl.peg

```diff
@@ -3528,9 +3594,73 @@ SetSvOption <- "setsv" EQUAL boolVal:Boolean {
     return mvColOptionArgs, nil
 }
 
+// returns *structs.QueryAggregators
+SPathBlock <- PIPE CMD_SPATH spathExpr:SPathArgumentsList? {
+    if spathExpr == nil {
+        return createSPathExpr("", "", "")
+    }
+    return spathExpr, nil
+}
+
+SPathArgumentsList <- SPACE first:SPathArgument rest:(SPACE SPathArgument)* {
+
+    restSlice := rest.([]any)
+
+    inputField, pathField, outputField := "", "", ""
+
+    numArgs := 1 + len(restSlice)
+
+    for i := 0; i < numArgs; i++ {
+        var arg string
+        if i == 0 {
+            arg = first.(string)
+        } else {
+            separator := restSlice[i - 1].([]any)
+            arg = separator[1].(string)
+        }
+
+        switch {
+        case strings.HasPrefix(arg, "input="):
+            inputField = strings.TrimPrefix(arg, "input=")
+        case strings.HasPrefix(arg, "output="):
+            outputField = strings.TrimPrefix(arg, "output=")
+        case strings.HasPrefix(arg, "path="):
+            pathField = strings.TrimPrefix(arg, "path=")
+        default:
+            return nil, fmt.Errorf("SPathArguments: Invalid Spath Argument: %v", arg)
+        }
+    }
+    
+    return createSPathExpr(inputField, pathField, outputField)
+}
+
+SPathArgument <- arg:(InputField / OutputField / PathField) {
+    return arg, nil
+}
+
+InputField <- "input" EQUAL field:FieldName {
+    return "input=" + field.(string), nil
+}
+OutputField <- "output" EQUAL field:FieldName {
+    return "output=" + field.(string), nil
+}
+PathField <- ("path" EQUAL SPathFieldString / SPathFieldString ) {
+     pathField := string(c.text)
+    if strings.HasPrefix(pathField, "path") {
+        pathField = strings.TrimSpace(strings.TrimPrefix(pathField, "path"))
+        pathField = strings.TrimSpace(strings.TrimPrefix(pathField, "="))
+        if pathField != "" && pathField[0] == '"' {
+            pathField = removeQuotes(pathField)
+        }
+    }
```

Comment:
I think instead of
```
    if strings.HasPrefix(pathField, "path") {
        pathField = strings.TrimSpace(strings.TrimPrefix(pathField, "path"))
        pathField = strings.TrimSpace(strings.TrimPrefix(pathField, "="))
```
You want
```
    if strings.HasPrefix(pathField, "path=") {
        pathField = strings.TrimSpace(strings.TrimPrefix(pathField, "path="))
```
That way if you parse something where `path=` was not given and the path starts with `"path"`, you don't get a bug. For example, `spath pathApple` should give the path as `pathApple` rather than just `Apple`

Reply:
yeah I did not think of this case. But also, I am thinking this will not handle the case where path is given like this:
`path = pathApple`. 
if there is space between path and `=`

Reply:
I could probably do this:

```go
        str = `path=  "pathval here"`
	str = strings.TrimSpace(strings.TrimPrefix(str, "path="))
	str = strings.TrimSpace(strings.TrimPrefix(str, "path ="))
	fmt.Println(str)
```

check for both the cases. Check the go playground: https://go.dev/play/p/iz8TfRN3LGi

Reply:
Is that a valid query? I would assume `spath path = myPath` is invalid, but do you see that in the documentation/examples/etc.?

Reply:
Hmm, as far as I see in the documentation it is mentioned as `path=` only. But I just thought users might type with space. I am not sure whether Splunk will accept this or throws an error.

But we defined `=` as this: `EQUAL <- EMPTY_OR_SPACE "=" EMPTY_OR_SPACE` 

And In many cases we use it like including for `eval`. so I assume even here this: `spath path = myPath` might be a valid case.

This is how eval syntax is mentioned.

```
eval
<field>=<expression> ["," <field>=<expression> ].
```

For Eval command in the splunk doc, they have given both kind of examples:

```
1. eval error = if(status == 200, "OK", "Problem")
2. eval velocity=distance/time
```
But for spath it was always `path=`

So should I change this to accept only `path=` format?

Reply:
Hmm ok then let's keep what you have so we can handle it with spaces

Reply:
Or actually, do something similar to what you suggested so you also handle the `pathApple` case

Reply:
Yes all the cases are being handled. All the below cases are working fine.

```
queries := []string{
		`* | spath pathApple`,
		`* | spath path= pathApple`,
		`* | spath path = pathApple`,
		`* | spath path =pathApple`,
	}
```